src/main/java/org/damap/base/rest/admin/service/RecommendedRepositoryService.java

```diff
@@ -0,0 +1,124 @@
+package org.damap.base.rest.admin.service;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.extern.jbosslog.JBossLog;
+import org.damap.base.domain.RecommendedRepository;
+import org.damap.base.r3data.RepositoriesService;
+import org.damap.base.rest.admin.domain.RecommendedRepositoryDO;
+import org.damap.base.rest.admin.mapper.RecommendedRepositoryDOMapper;
+
+@ApplicationScoped
+@JBossLog
+public class RecommendedRepositoryService {
+
+  @Inject RepositoriesService repositoriesService;
+
+  private boolean isValidRepositoryId(String repositoryId) {
+    if (repositoryId == null) {
+      return false;
+    }
+    if (!repositoryId.matches("^r3d\\d{6,}$")) {
+      return false;
+    }
+    return repositoriesService.getRepositoryDetailsFromRe3data(repositoryId) != null;
+  }
+
+  public List<RecommendedRepositoryDO> getRecommendedRepositories() {
+    List<RecommendedRepository> repositories = RecommendedRepository.list("ORDER BY id ASC");
+
+    return repositories.stream()
+        .map(
+            repo ->
+                RecommendedRepositoryDOMapper.mapEntityToDO(repo, new RecommendedRepositoryDO()))
+        .toList();
+  }
```

Comment:
```suggestion
  public List<RecommendedRepositoryDO> getRecommendedRepositories() {
    return RecommendedRepository.listAll().ascending();
  }
```

And at this point its just a wrapper around one line and we can remove the method.